### PR TITLE
feat: Generic Empty component 

### PR DESCRIPTION
**Pull request by shivani170**
# Description

To add a common empty state in
API Token List

Fixes [AB#1636](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/Devtron%20Team/Stories/?workitem=1636)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Create a new cluster in full mode/EA Only mode
- Go to API Token List with no token available & check empty state
- 

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/list/TriggerUrl.tsx [hunk 1]
```diff
@@ -188,14 +187,18 @@ export function CopyToClipboardText({ text, iconClass }: { text: string; iconCla
...
-            <EmptyState.Title>
-                <h4>{title || "No URLs available"}</h4>
-            </EmptyState.Title>
-            <EmptyState.Subtitle>{subtitle || "No URLs found in ingress and service resources"}</EmptyState.Subtitle>
-        </EmptyState>
+      <GenericEmptyState
+            title={title || "No URLs available"}
+            subTitle={subtitle || "No URLs found in ingress and service resources"}
+        />
+        // <EmptyState>
```
Comment (vivek-devtron) [inline src/components/app/list/TriggerUrl.tsx:194]: please remove commented code

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 2]
```diff
@@ -167,14 +172,16 @@ export default function CDDetails() {
                             />
                         </Route>
                     ) : !envId ? (
-                        <EmptyView
-                            title="No environment selected"
-                            subTitle="Please select an environment to start seeing CD deployments."
+                        <GenericEmptyState
+                            image={AppNotDeployed}
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/CDDetails.tsx:176]: We can remove image from here as this is the default image

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 3]
```diff
@@ -167,14 +172,16 @@ export default function CDDetails() {
...
+                            title={EMPTY_STATE_STATUS.CD_DETAILS_ENVIRONMENT.TITLE}
+                            subTitle={EMPTY_STATE_STATUS.CD_DETAILS_ENVIRONMENT.SUBTITLE}
                         />
                     ) : (
-                        <EmptyView
-                            title="No deployments"
-                            subTitle={`No deployment history available for the ${environment?.environmentName} environment.`}
+                        <GenericEmptyState
+                            image={AppNotDeployed}
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/CDDetails.tsx:182]: same here

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 4]
```diff
@@ -314,8 +317,11 @@ function ChartDeploymentHistory({
                 (selectedDeploymentTabIndex === 2 && !selectedDeploymentManifestDetail.manifest))
         ) {
             return (
-                <CDEmptyState
-                    subtitle={`${deploymentTabs[selectedDeploymentTabIndex]} is not available for this deployment`}
+                <GenericEmptyState
+                    image={AppNotDeployed}
```
Comment (vivek-devtron) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:321]: same here

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx [hunk 5]
```diff
@@ -551,7 +557,12 @@ function ChartDeploymentHistory({
             )
         } else if (!deploymentHistoryArr || deploymentHistoryArr.length <= 0) {
             return (
-                <CDEmptyState subtitle="Data for previous deployments is not available. History for any new deployment will be available here." />
+                <GenericEmptyState
+                    image={AppNotDeployed}
```
Comment (vivek-devtron) [inline src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx:561]: same here

File: src/components/app/list/TriggerUrl.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/app/list/TriggerUrl.tsx]: done

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 2 above)
Comment (shivani170) [reply src/components/app/details/cdDetails/CDDetails.tsx]: Done

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 4 above)
Comment (shivani170) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: Done

File: src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx (hunk 5 above)
Comment (shivani170) [reply src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx]: Done

**Comment by shivani170** [comment]
Duplicate of https://github.com/devtron-labs/dashboard/pull/965

**Pull request closed without merging**